src/main/java/org/springframework/data/couchbase/repository/TransactionResultKey.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.repository;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.data.annotation.QueryAnnotation;
+
+/**
+ * Indicates the field should hold key to lookup the TransactionGetResult and should NOT be considered part of the
+ * document.
+ *
+ * @author Michael Reiche
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD })
+@Documented
+@QueryAnnotation
+public @interface TransactionResultKey {
```

Comment:
My suggestion would be a slightly different name - the "key" is an implementation detail? Maybe something more towards TransactionMeta (so that also in the future we can change the content etc).

Reply:
Renamed to TransactionMeta.